# ãƒ‡ã‚£ãƒ¬ã‚¯ã‚¿ãƒ¼ï¼ˆDirectorï¼›ç›£ç£ï¼‰
#
# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ã®ä½¿ã„æ–¹ã®å®Ÿä¾‹ã§ã™ã€‚ã“ã®ãƒãƒ¼ãƒ‰è‡ªèº«ã¯ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°æœ¬ä½“ã§ã¯ã‚ã‚Šã¾ã›ã‚“
#
extends Node2D


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# å†…ãƒ‘ã‚¹é–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# ã‚¤ãƒ©ã‚¹ãƒˆãƒ¬ãƒ¼ã‚¿ãƒ¼å–å¾—
func get_illustrator():
	return $"ğŸŒIllustrator"


# ãƒ—ãƒ­ã‚°ãƒ©ãƒ ã‚ºãƒ»ãƒãƒ–å–å¾—
func get_programs_hub():
	return $"ğŸŒProgrammer/ğŸ›©ï¸Hub"


# ãƒ†ãƒ­ãƒƒãƒ—ãƒ»ã‚³ãƒ¼ãƒ‡ã‚£ãƒãƒ¼ã‚¿ãƒ¼å–å¾—
func get_telop_coordinator():
	return $"ğŸŒTelopCoordinator"


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãã®ä»–
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# Called when the node enters the scene tree for the first time.
func _ready():

	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	# è¡¨ç¤º
	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

	# ãƒ†ãƒ­ãƒƒãƒ—ãƒ»ã‚³ãƒ¼ãƒ‡ã‚£ãƒãƒ¼ã‚¿ãƒ¼è¡¨ç¤º
	self.get_telop_coordinator().show()

	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	# åˆæœŸåŒ–
	# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
	$"ğŸ›©ï¸KeyConfigHub".entry()


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ã§å¿…è¦ã§ã™
	$"ğŸ›©ï¸KeyConfigHub".on_process(delta)


func _unhandled_input(event):
	# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ã§å¿…è¦ã§ã™
	$"ğŸ›©ï¸KeyConfigHub".on_unhandled_input(event)


# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°é–‹å§‹æ™‚
func on_key_config_entered():
	# èƒŒæ™¯
	self.get_illustrator().visible = true
	self.get_programs_hub().images.find_node("ğŸ—»å´å·é§…å‰").visible = true


# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°å®Œäº†æ™‚
func on_key_config_exited():
	print("ï¼»ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ï¼½ã€€å®Œäº†")

	# èƒŒæ™¯
	self.get_programs_hub().images.find_node("ğŸ—»å´å·é§…å‰").visible = false
